src/plugins/vis_type_vega/public/data_model/search_api.ts

```diff
@@ -61,6 +61,7 @@ export class SearchAPI {
     return combineLatest(
       searchRequests.map((request) => {
         const requestId = request.name;
+        const dataSourceId = request.data_source_id;
```

Comment:
Can we add test coverage? thanks

Reply:
Added some test cases to parse the id from the `data_source_name`